lib/iris/cube.py

```diff
@@ -880,8 +880,11 @@ def convert_units(self, unit):
 
         """
         # If the cube has units convert the data.
-        if not self.units.is_unknown():
-            self.data = self.units.convert(self.data, unit)
+        if self.units.is_unknown():
+            raise iris.exceptions.UnitConversionError(
+                'cannot convert from unknown units to {!s}, '
+                'set the "units" attribute instead'.format(unit))
```

Comment:
> What about something like 'Cannot convert from unknown units. The "cube.units" attribute may be set directly.'

@rcomer I like that - it is clear and descriptive without being complicated. If you could change the error message to that it would be great :+1: